### DISCUSSION TITLE
DoH unreliable on EdgeRouter X (mipsle) router

### DISCUSSION THREAD

**Discussion by BrianG61UK** [DNSCrypt/dnscrypt-proxy#1998] [Potential issues]
When using DoH I find that after some time (typically less than 24 hours) I find I'm unable to resolve any DNS queries and the dnscrypt-proxy process is consuming 100++% CPU. No problem when using DNSCrypt rather than DoH.

A similar problem with [nextdns/nextdns](https://github.com/nextdns/nextdns) cli was fixed by reverting from go 1.17 to the latest go 1.16.

**Comment by hugepants** [comment]
I have experienced this with the ER-X on OpenWrt. Related issues: https://github.com/DNSCrypt/dnscrypt-proxy/issues/797 and https://github.com/DNSCrypt/dnscrypt-proxy/issues/794

I was not able to solve the problem for DoH after trying different `tls_cipher_suite` values, so my solution was to use only DNSCrypt resolvers.

That is interesting about [reverting Go to 1.16](https://github.com/nextdns/nextdns/pull/618), I don't know if this is feasible in dnscrypt-proxy, but I hope they can identify the root cause.

**Reply by BrianG61UK** [reply]
Thanks. In particular, thanks for the useful links.

**Comment by BrianG61UK** [comment]
Nothing seems to be happening about this :(

**Reply by lifenjoiner** [reply]
We need someone to help locate exactly where the problem it is ...
It seems to be an issue with go.
Maybe try go v1.18+.

**Reply by BrianG61UK** [reply]
Myself, I have no idea how to build dnscrypt-proxy, let alone how to build it with different versions of go.

**Reply by lifenjoiner** [reply]
~~I just built 2 versions by go v1.18.3 with github CI, that also have mipsle builds. If you are interested, try from:
[TestAndroid-3](https://github.com/lifenjoiner/dnscrypt-proxy/releases/tag/TestAndroid-3)~~

Update:
Now, [Release 2.1.2](https://github.com/DNSCrypt/dnscrypt-proxy/releases/tag/2.1.2) is built with go v1.18.4.

**Reply by hugepants** [reply]
@BrianG61UK
For me this issue is solved in dnscrypt-proxy 2.1.2 (Go 1.18.4), can you please try it and let us know your results?

And for the best DoH performance on MIPS, you should use a server that supports TLSv1.3 (304) and offers a `TLS_CHACHA20_POLY1305_SHA256 (4867)` cipher, which many do these days. Cloudflare, Google, NextDNS all do. You can check this by setting `log_level = 1` in `dnscrypt-proxy.toml`.

**Reply by BrianG61UK** [reply]
@hugepants Okay. I'm now running 2.1.2 and DoH ('cisco-doh') instead of my usual DNSCrypt ('cisco').
I'll see how I get on over the next few days.

**Reply by BrianG61UK** [reply]
Seems like it must be fixed. No problems and it typically lasted less than a day when the bug was present.

**Reply by hugepants** [reply]
Excellent, thanks for the update.
